pageserver/src/tenant_mgr.rs

```diff
@@ -422,9 +270,6 @@ pub async fn delete_timeline(tenant_id: TenantId, timeline_id: TimelineId) -> an
     match get_tenant(tenant_id, true) {
         Ok(tenant) => {
             tenant.delete_timeline(timeline_id)?;
```

Comment:
Wouldn't we need to shut down the tasks of the timeline _after_ we transition it to `::Paused` state?

For example, the `"initial size calculation"` task doesn't use `task_mgr::is_shutdown_requested`, it just reacts to timeline state changes.

Reply:
Agreed, this is really shoddy. It's not new with this PR nor #2785, though. I'll create a separate issue to clean that up.